e2e/libs/node/node.py

```diff
@@ -17,53 +18,48 @@ def __init__(self):
         self.aws_client = boto3.client('ec2')
         #logging.warn(f"describe_instances = {self.aws_client.describe_instances()}")
 
-    def restart_all_nodes(self):
+    def reboot_all_nodes(self, shut_down_time_in_sec=60):
         instance_ids = [value for value in self.mapping.values()]
         print(instance_ids)
         resp = self.aws_client.stop_instances(InstanceIds=instance_ids)
         print(resp)
         waiter = self.aws_client.get_waiter('instance_stopped')
         waiter.wait(InstanceIds=instance_ids)
         print(f"all instances stopped")
-        time.sleep(60)
+        time.sleep(shut_down_time_in_sec)
         resp = self.aws_client.start_instances(InstanceIds=instance_ids)
         print(resp)
         waiter = self.aws_client.get_waiter('instance_running')
         waiter.wait(InstanceIds=instance_ids)
         wait_for_cluster_ready()
         print(f"all instances running")
 
-    def reboot_node(self, running_on_node_name, reboot_node_name, shut_down_time_in_sec=10):
-        with open('/tmp/instance_mapping', 'r') as f:
-            mapping = yaml.safe_load(f)
-            reboot_node_id = mapping[reboot_node_name]
-
-        filepath = './templates/litmus/reboot-node.yaml'
-        with open(filepath, 'r') as f:
-            data = yaml.safe_load(f)
-            data['spec']['components']['runner']['nodeSelector']['kubernetes.io/hostname'] = running_on_node_name
-            data['spec']['experiments'][0]['spec']['components']['nodeSelector']['kubernetes.io/hostname'] = running_on_node_name
-            data['spec']['experiments'][0]['spec']['components']['env'][1]['value'] = str(shut_down_time_in_sec)
-            data['spec']['experiments'][0]['spec']['components']['env'][2]['value'] = reboot_node_id
-
-        with open(filepath, 'w') as file:
-            yaml.dump(data,file,sort_keys=False)
-
-        apply_cr_from_yaml(filepath)
+    def reboot_node(self, reboot_node_name, shut_down_time_in_sec=60):
+        instance_ids = [self.mapping[reboot_node_name]]
+        print(instance_ids)
+        resp = self.aws_client.stop_instances(InstanceIds=instance_ids)
```

Comment:
You can abstract the underling call, so later it will be supported for different implementation for various infras. This can be tackled when you work on running tests on local or harvester infra.

Reply:
Tackle this later when working on local environment implementation. I mainly use cloud providers for testing. I'll try to setup a local environment later to have a clearer picture of what it would look like.